bolt/lib/Core/BinaryContext.cpp

```diff
@@ -1624,10 +1624,26 @@ DWARFContext *BinaryContext::getDWOContext() const {
   return &DWOCUs.begin()->second->getContext();
 }
 
+bool BinaryContext::isValidDwarfUnit(DWARFUnit *const DU) const {
+  // Invalid DWARF unit with a DWOId but lacking a dwo_name.
+  if (DU->getDWOId() && !DU->isDWOUnit() &&
+      !DU->getUnitDIE().find(
+          {dwarf::DW_AT_dwo_name, dwarf::DW_AT_GNU_dwo_name})) {
+    this->outs() << "BOLT-ERROR: Broken DWARF found in CU at offset 0x"
+                 << Twine::utohexstr(DU->getOffset()) << " (DWOId=0x"
+                 << Twine::utohexstr(*(DU->getDWOId()))
+                 << ", missing DW_AT_dwo_name / DW_AT_GNU_dwo_name).\n";
```

Comment:
nit: for message formatting, start with lowercase (s/Broken/broken) and drop the trailing period.